sdk/cosmos/azure-spring-data-cosmos-test/src/test/java/com/azure/spring/data/cosmos/core/ReactiveCosmosTemplateIT.java

```diff
@@ -281,6 +308,35 @@ public void testUpsert() {
         assertThat(responseDiagnosticsTestUtils.getCosmosDiagnostics()).isNotNull();
     }
 
+    @Test
+    public void testPatch() {
+        final Mono<Person> patch = cosmosTemplate.patch(insertedPerson, operations, null);
+        StepVerifier.create(patch).expectNextCount(1).verifyComplete();
+        Mono<Person> patchedPerson = cosmosTemplate.findById(containerName, insertedPerson.getId(), Person.class);
+        StepVerifier.create(patchedPerson).expectNextCount(1).verifyComplete();
+        assertEquals(patchedPerson.block().getAge(), PATCH_AGE_1);
+    }
+
+    @Test
+    public void testPatchMultiOperations() {
+        final Mono<Person> patch = cosmosTemplate.patch(insertedPerson, multiPatchOperations, null);
+        StepVerifier.create(patch).expectNextCount(1).verifyComplete();
+        Mono<Person> patchedPerson = cosmosTemplate.findById(containerName, insertedPerson.getId(), Person.class);
+        StepVerifier.create(patchedPerson).expectNextCount(1).verifyComplete();
+        assertEquals(patchedPerson.block().getAge().intValue(), (AGE + PATCH_AGE_INCREMENT));
+        assertEquals(patchedPerson.block().getHobbies(),PATCH_HOBBIES);
+        assertEquals(patchedPerson.block().getFirstName(), PATCH_FIRST_NAME);
+        assertEquals(patchedPerson.block().getPassportIdsByCountry(), NEW_PASSPORT_IDS_BY_COUNTRY);
+    }
+
+    @Test
+    public void testPatchPreConditionFail() {
```

Comment:
should we also add patch precondition successful test case?

Reply:
Added